src/phase0/state_transition/block_processing.rs

```diff
@@ -236,16 +236,28 @@ fn process_eth1_data<
         MAX_VALIDATORS_PER_COMMITTEE,
         PENDING_ATTESTATIONS_BOUND,
     >,
-    _body: &BeaconBlockBody<
+    body: &BeaconBlockBody<
         MAX_PROPOSER_SLASHINGS,
         MAX_VALIDATORS_PER_COMMITTEE,
         MAX_ATTESTER_SLASHINGS,
         MAX_ATTESTATIONS,
         MAX_DEPOSITS,
         MAX_VOLUNTARY_EXITS,
     >,
-    _context: &Context,
+    context: &Context,
 ) -> Result<(), Error> {
+    state.eth1_data_votes.push(body.eth1_data.clone());
+
+    let votes_count = state
+        .eth1_data_votes
+        .iter()
+        .filter(|&vote| *vote == body.eth1_data)
```

Comment:
nitpick: 

i believe we can just do this
```suggestion
        .filter(|vote| *vote == body.eth1_data)
```

and there are sufficient blanket impls of `PartialEq` for refs that the compiler will do the right thing

and then we have less code to grok

Reply:
The issue is 'body.eth1_data' is not a ref. 

I could do
```suggestion
        .filter(|vote| **vote == body.eth1_data)
```
or
```suggestion
        .filter(|vote| *vote == &body.eth1_data)
```
but would that change anything?

Reply:
ah good point, i was confused about the `body` item!